common/cache_managers.py

```
@@ -0,0 +1,565 @@
+            return None
+
+    def invalidate_cache(self, reason: str = "manual") -> None:
+        """Invalidate both module-level and Redis cache."""
+        with self._cache_lock:
+            super().invalidate_cache(reason)
+            self._module_cache.clear()
+
+    def _register_signals(self):
+        """Register signal handlers for automatic cache invalidation."""
+
+        def handle_model_change(sender, instance, **kwargs):
```

Ruff (0.12.2)

Missing return type annotation for private function `handle_model_change`

Add return type annotation: `None`

(ANN202)